File: test_samples/test_sample.js
function simpleFunction() {
    console.log("Hello JavaScript");
}

function functionWithIf(x) {
    if (x > 0) {
        return x + 1;
    } else {
        return 0;
    }
}

function complexFunction(a, b) {
    if (a > b) {
        if (a > 10) {
            return a * 2;
        } else {
            return a + b;
        }
    } else {
        if (b > 10) {
            return b * 3;
        } else {
            return a - b;
        }
    }
}

const arrowFunction = (x, y) => {
    if (x && y) {
        return x * y;
    }
    return 0;
};

File: test_sample.rs
fn simple_function() {
    println!("Hello world");
}

fn function_with_if(x: i32) -> i32 {
    if x > 0 {
        x + 1
    } else {
        0
    }
}

fn complex_function(a: i32, b: i32) -> i32 {
    if a > b {
        if a > 10 {
            a * 2
        } else {
            a + b
        }
    } else {
        if b > 10 {
            b * 3
        } else {
            a - b
        }
    }
}